src/masks/InteractionMasks.tsx

```diff
@@ -140,53 +129,85 @@ export default function InteractionMasks<R>({
     };
   }
 
-  function getNextPosition(keyCode: number) {
-    switch (keyCode) {
-      case KeyCodes.ArrowUp:
-        return { ...selectedPosition, rowIdx: selectedPosition.rowIdx - 1 };
-      case KeyCodes.ArrowDown:
-        return { ...selectedPosition, rowIdx: selectedPosition.rowIdx + 1 };
-      case KeyCodes.ArrowLeft:
-        return { ...selectedPosition, idx: selectedPosition.idx - 1 };
-      case KeyCodes.ArrowRight:
-        return { ...selectedPosition, idx: selectedPosition.idx + 1 };
+  function getNextPosition(key: string, mode = cellNavigationMode, shiftKey = false) {
+    const { idx, rowIdx } = selectedPosition;
+    let nextPosition: Position;
+    switch (key) {
+      case 'ArrowUp':
+        nextPosition = { idx, rowIdx: rowIdx - 1 };
+        break;
+      case 'ArrowDown':
+        nextPosition = { idx, rowIdx: rowIdx + 1 };
+        break;
+      case 'ArrowLeft':
+        nextPosition = { idx: idx - 1, rowIdx };
+        break;
+      case 'ArrowRight':
+        nextPosition = { idx: idx + 1, rowIdx };
+        break;
+      case 'Tab':
+        nextPosition = shiftKey ? { idx: idx - 1, rowIdx } : { idx: idx + 1, rowIdx };
```

Comment:
```suggestion
        nextPosition = { idx: idx + (shiftKey ?  -1 : 1), rowIdx };
```

Reply:
Also shouldn't this handle cellNavigationMode?

Reply:
It is handled by the mode param

https://github.com/adazzle/react-data-grid/pull/1941/files#diff-63a962509ad28f9bf4dbc6c8a1c50569R233